### PR TITLE
feat: Make express mode default

### LINKED ISSUE #15743: Make express build the default mode
- Remove feature flag from express mode making it the default one.
- Add parameter `frontend.hotdeploy` for disabling express mode
- Remove enableDevServer

### PR DESCRIPTION
**Pull request by caalador**
Set express mode as default mode
for development time.

Closes #15743

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/AbstractConfiguration.java [hunk 1]
```diff
@@ -41,16 +41,16 @@ public interface AbstractConfiguration extends Serializable {
     boolean isProductionMode();
 
     /**
-     * Get if the dev server should be enabled. True by default
+     * Get if the dev server should be enabled. false by default as express mode
+     * should be used.
      *
      * @return true if dev server should be used
      */
     default boolean enableDevServer() {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/AbstractConfiguration.java:49]: Don't we want to remove `enableDevServer` completely? Or do we keep this method just to not make a breaking changes?

File: flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java [hunk 2]
```diff
@@ -373,4 +382,9 @@ private void checkSendUrlsAsParameters() {
                 DEFAULT_SEND_URLS_AS_PARAMETERS);
     }
 
+    private void checkFrontendHotdeploy() {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:385]: Does that mean that setting `enableDevServer=true` won't take any effect now?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:385]: It's `frontend.hotdeploy`, but tokenFile is read for the old value also.

File: flow-server/src/main/java/com/vaadin/flow/server/AbstractConfiguration.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/AbstractConfiguration.java:49]: I would suggest changing the method (and properrty) name in its own PR

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/AbstractConfiguration.java:49]: Ok, sounds good

File: flow-server/src/main/java/com/vaadin/flow/server/startup/AbstractConfigurationFactory.java [hunk 3]
```diff
@@ -125,8 +126,11 @@ protected Map<String, String> getConfigParametersUsingTokenData(
         // These should be internal only so if there is a System
         // property override then the user probably knows what
         // they are doing.
-        if (buildInfo.hasKey(SERVLET_PARAMETER_ENABLE_DEV_SERVER)) {
-            params.put(SERVLET_PARAMETER_ENABLE_DEV_SERVER, String.valueOf(
+        if (buildInfo.hasKey(FRONTEND_HOTDEPLOY)) {
+            params.put(FRONTEND_HOTDEPLOY,
+                    String.valueOf(buildInfo.getBoolean(FRONTEND_HOTDEPLOY)));
+        } else if (buildInfo.hasKey(SERVLET_PARAMETER_ENABLE_DEV_SERVER)) {
+            params.put(FRONTEND_HOTDEPLOY, String.valueOf(
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/startup/AbstractConfigurationFactory.java:133]: Ok, here is the place where it read enableDevServer from token file. Thanks

File: flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java [hunk 4]
```diff
@@ -516,7 +516,7 @@ public void externalStatsFileTrue_predefinedValuesAreNotOverridden()
                 .singletonMap(PARAM_TOKEN_FILE, tokenFile.getPath()));
 
         assertEquals(true, config.isProductionMode());
-        assertEquals(true, config.enableDevServer());
+        assertEquals(false, config.enableDevServer());
```
Comment (mshabarov) [inline flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java:519]: Does it makes sense to add one more check for hotdeploy new parameter?

Comment (caalador) [reply flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java]: Updated the test.

File: flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java [hunk 5]
```diff
@@ -498,26 +498,25 @@ private VaadinConfig mockTokenFileViaContextParam(String content)
...
                         "\"externalStatsFile\": true", "}"));
 
         DeploymentConfiguration config = createConfig(Collections
                 .singletonMap(PARAM_TOKEN_FILE, tokenFile.getPath()));
 
-        assertEquals(true, config.isProductionMode());
-        assertEquals(false, config.enableDevServer());
-        assertEquals(true, config.isStatsExternal());
+        assertTrue(config.isProductionMode());
+        assertFalse("Dev server should be default false due to stats",
```
Comment (mshabarov) [inline flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java:517]: Not sure I follow: if the frontend.hotdeploy is set to true in the stats, it should be also true for enableDevServer?

Comment (caalador) [reply flow-server/src/test/java/com/vaadin/flow/server/DeploymentConfigurationFactoryTest.java:517]: if statsExternal is set to true then no other configuration is set from the token file.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**